torch/utils/backend_registration.py

```diff
@@ -104,6 +104,12 @@ def _get_current_device_index():
             device_idx = _get_current_device_index()
         else:
             device_idx = device.index
+    # if isinstance(device, str), this means that the parameter passed in is in the string format "foo:0"
+    elif isinstance(device, str):
+        if device.split(":")[0] == custom_backend_name:
```

Comment:
Actually, hold on, what about `device="cuda"` or device="foo". Do you need to add error handing for that case as well?